src/App.tsx

```diff
@@ -15,48 +16,127 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((prevGood, currGood) => {
+      switch (sortType) {
+        case SortType.ALPHABET:
+          return prevGood.localeCompare(currGood);
+        case SortType.LENGTH:
+          return prevGood.length - currGood.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const visibleGoods = getReorderedGoods(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
+  const alphaSortHandler = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const lengthSortHandler = () => {
+    setSortType(SortType.LENGTH);
+  };
```

Comment:
You can combine this into one function with argument. Or use set functions directly.